smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -257,18 +336,22 @@ contract BridgeBank is BankStorage,
         }
     }
 
-    /*
-     * @dev: Burns BridgeTokens representing native Cosmos assets.
-     *
-     * @param recipient: bytes representation of destination address.
-     * @param token: token address in origin chain (0x0 if ethereum)
-     * @param amount: value of deposit
+    /**
+     * @notice Burns `amount` `token` tokens for `recipient`
+     * @dev Burns BridgeTokens representing native Cosmos assets
+     * @param recipient Bytes representation of destination address
+     * @param token Token address in origin chain (0x0 if ethereum)
+     * @param amount How much will be burned
      */
     function burn(
         bytes calldata recipient,
         address token,
         uint256 amount
     ) external validSifAddress(recipient) onlyCosmosTokenWhiteList(token) whenNotPaused {
+        // revert if the token doesn't have a denom
+        string memory denom = getDenom(token);
+        require(keccak256(abi.encodePacked(denom)) != keccak256(abi.encodePacked("")), "INV_DENOM");
```

Comment:
Not sure we should hard assert this. I think it would be better to just ensure that things are in the whitelist

Reply:
Is there a situation where a token that's in the whitelist doesn't have a denom?
If it's added 'manually' through `addExistingBridgeToken`, it might not have a denom (nothing enforces it right now).

Reply:
As discussed via Slack:
"In the case that the tokens do not have a cosmosDenom, but are still in the whitelist, they should be burned"
I'll implement it.